### PR TITLE
synset pos parameter

### PR DESCRIPTION
**Pull request by it176131**
added `pos` parameter to `.synsets(...)` method so user can filter `wordnet.synsets(...)` directly

### CODE REVIEW

**Comment by it176131** [comment]
Hi @frascuchon 
Any word on when this will be accepted/rejected?

File: spacy_wordnet/wordnet_domains.py
Comment (frascuchon) [inline spacy_wordnet/wordnet_domains.py]:
```suggestion
    def synsets(self, pos: Optional[Union[str, List[str]]] = None) -> List[Synset]:
```

File: spacy_wordnet/wordnet_domains.py
Comment (frascuchon) [inline spacy_wordnet/wordnet_domains.py]:
```suggestion
    def __find_synsets(token: Token,
                       lang: str,
                       pos: Optional[Union[str, List[str]]] = None) -> List[Synset]:
```

**Review by frascuchon — COMMENTED** [review]
Great work @it176131 !!!

Just a few code-style suggestions. Once they are resolve, we can merge the PR 

Thanks!

**Comment by it176131** [comment]
> Great work @it176131 !!!
> 
> Just a few code-style suggestions. Once they are resolve, we can merge the PR
> 
> Thanks!

Fixed! Thanks for the suggestions 🙂

**Comment by frascuchon** [comment]
Sorry @it176131 

I've merged some older work. Could you resolve conflicts in your PR?

**Comment by it176131** [comment]
> Sorry @it176131
> 
> I've merged some older work. Could you resolve conflicts in your PR?

Resolved!

**Review by frascuchon — APPROVED** [review]

File: spacy_wordnet/wordnet_domains.py
Comment (frascuchon) [inline spacy_wordnet/wordnet_domains.py:116]: @it176131 Is there any reason to not search by the `token.pos` no `pos` param provided?

Comment (it176131) [reply spacy_wordnet/wordnet_domains.py:116]: > @it176131 Is there any reason to not search by the token.pos no pos param provided?

Not sure I understand. Are you saying that I can provide a list as an argument to the `spacy2wordnet_pos` func? Or are you asking why I don't limit strictly to the `token.pos`?

If the latter, I wanted the user to have the ability to filter to multiple `pos`, not just the `token.pos`. As an example, if the `token.pos` is `VERB`, but the user wants `VERB` and `NOUN`, then I'd need to provide both options rather than the single.

Of course if the `spacy2wordnet_pos` function can take a list, then I can do away with the loop. But I didn't think it could.

Does that answer the question or am I missing the mark?

Comment (frascuchon) [reply spacy_wordnet/wordnet_domains.py:116]: What I mean is that, if the user does not provide the `pos` parameter, the wordnet search will use all available pos for finding synset.

My proposal is just to keep the old behavior and set as default the `token.pos` value.

I will add a code suggestion to implement my proposal.

File: spacy_wordnet/wordnet_domains.py
Comment (frascuchon) [inline spacy_wordnet/wordnet_domains.py]:
```suggestion
    def __find_synsets(token: Token,
                       lang: str,
                       pos: Optional[Union[str, List[str]]] = None) -> List[Synset]:
        if pos is None:
            pos = []
        elif isinstance(pos, str):
            pos = [pos]
        elif not isinstance(pos, list):
            try:
                pos = list(pos)
            except TypeError:
                raise TypeError("pos argument must be None, type str, or type list.")
        
        acceptable_pos = {"verb": VERB, "noun": NOUN, "adj": ADJ} # We can define this as a private class constant
        # check if any element in `pos` is not in `acceptable_pos`
        if set(pos).difference(acceptable_pos):
            raise ValueError("pos argument must be a combination of 'verb', "
                             "'noun', or 'adj'.")

        token_pos: List[int] = [acceptable_pos[k] for k in pos]
        if not token_pos:
           token_pos = [token.pos]
```

Comment (it176131) [reply spacy_wordnet/wordnet_domains.py]: Okay I understand now! I can get behind this suggestion. I'm going to update the docstring and unit test.

**Comment by it176131** [comment]
@frascuchon I've updated the docstring and added a couple assertion checks in the the unit test (one for when `pos` arg is a list of values and one for when the arg is `None`).

Waiting for your review

File: spacy_wordnet/wordnet_domains.py
Comment (frascuchon) [inline spacy_wordnet/wordnet_domains.py]: I've seen a problem here:

If no `pos` param is provided, and `token.pos` is used as default, the `token.lemma_` is always included as a word variant. This could cause a weird behavior of the component

```suggestion
        if token.pos in (token_pos if pos else acceptable_pos.values()):
```

Comment (it176131) [reply spacy_wordnet/wordnet_domains.py]: Let me make sure I understand correctly.

If `pos is None` and `token.pos` is something other than `NOUN`, `VERB`, or `ADJ` --> this may cause unexpected `token.lemma_` values to be appended to the `word_variants` list.

# EDIT
I tested with `token = nlp("quickly")[0]` before implementing your suggestion:

``` python
token = nlp("quickly")[0]

synsets = token._.wordnet.synsets(pos=None)

set(synsets)

>>> {Synset('cursorily.r.01'), Synset('promptly.r.01'), Synset('quickly.r.01')}
```

I agree this is unexpected.

**Comment by it176131** [comment]
@frascuchon I'm going to update the docstring to reflect the recent changes and add an edge case to the unit test.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**
